ferry.go

```diff
@@ -395,30 +430,30 @@ func (f *Ferry) FlushBinlogAndStopStreaming() {
 			f.ErrorHandler.Fatal("wait_replica", err)
 		}
 
-		err = f.WaitUntilReplicaIsCaughtUpToMaster.Wait()
 		if err != nil {
 			f.ErrorHandler.Fatal("wait_replica", err)
 		}
 	}
 
-	f.BinlogStreamer.FlushAndStop()
+	f.BinlogStreamer.FlushAndStop(ctx)
```

Comment:
Again, this seems super dangerous. The method should either return an error (that can be `context.Canceled`or nil) or just not allow interruption at this stage.